pro_tes/middleware/task_distribution/distance.py

```diff
@@ -1,42 +1,25 @@
-"""Module for task distribution logic."""
+"""Module for distance based task distribution logic ."""
 
 from copy import deepcopy
-import random
-from urllib.parse import urlparse
-from socket import gaierror, gethostbyname
-from typing import Dict, List, Set, Tuple, Optional
 from itertools import combinations
-from ip2geotools.errors import InvalidRequestError
-from ip2geotools.databases.noncommercial import DbIpCity
-from geopy.distance import geodesic
+from socket import gaierror, gethostbyname
+from typing import Dict, List, Optional, Set, Tuple
+from urllib.parse import urlparse
 
 from flask import current_app
-import requests
-
-from pro_tes.middleware.models import AccessUriCombination, TesUriList
+from geopy.distance import geodesic
+from ip2geotools.databases.noncommercial import DbIpCity
+from ip2geotools.errors import InvalidRequestError
 
+from pro_tes.middleware.models import (
+    AccessUriCombination,
+    TaskParams,
+    TesDeployment,
+    TesStats
+)
 
+# pylint: disable-msg=R0912
 # pylint: disable-msg=too-many-locals
-def random_task_distribution() -> Optional[str]:
```

Comment:
Please use expressive Pylint rules (`too-many-branches` better than `R0912`).

Reply:
By the way, rules such as `too-many-branches` and `too-many-locals` are sensible rules. They may be cases where it makes sense to disable, e.g., if you have a clean, specialized function that goes through lots of conditions. But in _most_ cases, I think what they should suggest to you to carefully check your code for smells:
* Is every local _really_ needed?
* Is every branch _really_ needed?
* If both answers are yes: Could the function perhaps be refactored into multiple subfunctions? Or perhaps a class-based structure?

Reply:
Understood will try to eliminate these errors rather than disabling them.